### ISSUE TITLE
Invoke-CMApplyDriverPackage.ps1 v2.1.2 not detecting (dell) Driver package

### ISSUE THREAD

**Issue by MaartenPauchet** [SCConfigMgr/ConfigMgr#68]
Hi,

updated my setup yesterday, new Driver Download 6.1, used the action to rename my HP driver packages for the short name version. 
Updated the CMApplyDriverPackage to v2.1.2.
I tried to load a Dell Precision 7520 today and it's driver pack is not detected. 
I checked the ApplyDriverPackage.log:
``===== SCConfigMgr Invoke-CMApplyDriverPackage Version 2.1.2 ====
Driver download package process initiated
Manufacturer determined as: Dell
Computer model determined as: Precision 7520
Computer SKU determined as: 07B0
Fallback SKU determined as: 07B0
Retrieved a total of 84 driver packages from web service
Attempting to detect OS Image data from task sequence, running in DeploymentType: BareMetal
Translated OSName from OSImageVersion: Windows 10
Translated OSImageArchitecture: x64
Translated OS image version from '10.0.16299.431' to: 
Using computer detection method: SystemSKU
<skipping a lot of mismatching driver packs here>
Attempting to find a match for driver package: Drivers - Dell Precision 7520 - Windows 10 x64 (xxx001C3)
Unable to match computer model using detection method: SystemSKU (07B0)
Fallback from SystemSKU match found for computer model instead using detection method: SystemSKU (Precision 7520)
Driver package does not meet computer model, manufacturer and operating system and architecture criteria: Drivers - Dell Precision 7520 - Windows 10 x64 (MOW001C3)
Computer model detection logic failed for all detected driver packages
``
so it looks like it's no longer finding my driver packages for a Dell device which was working fine with the v2.1.0 I had before (exact same test machine). My bet would be on the line: 
Translated OS image version from '10.0.16299.431' to:
 
which doesn't seem to know what short version to choose (should 1709) because the hash table in your source only appears to have entries for : 
# Windows Version Hash Table
	$WindowsBuildHashTable = @{
		"1803"  = "10.0.17134.1"
		"1709"  = "10.0.16299.15"
		"1703"  = "10.0.15063.0"
                "1607" = "10.0.14393.0"

hope this helps to figure out what went wrong and make the driver management solution even better!

**Comment by MaartenPauchet** [comment]
just reverted my Invoke-CMApplyDriverPackage.ps1 back to 2.1.1 > still fails.
reverted it back to 2.1.0 and it works again...

**Comment by NickolajA** [comment]
Hi,

This will be resolved in 2.1.3 where the table part will be converted to a switch statement.

**Comment by MaartenPauchet** [comment]
I repeated my tests today after upgrading my file from 2.1.0 > 2.1.3 and I continue to have the same issue.
I did the test another time after also renaming my Dell driver package to include the "1709" in the name. 
unfortunately that still did not make a difference. I've attached my log file. 
I also used the script to install a Windows 7 (on a latitude E6530)  and that one installed fine. 

[ApplyDriverPackage.log](https://github.com/SCConfigMgr/ConfigMgr/files/2352205/ApplyDriverPackage.log)
